private/bufpkg/bufconnect/interceptors.go

```diff
@@ -82,25 +71,66 @@ func NewAuthorizationInterceptorProvider(container app.EnvContainer) func(string
 	}
 }
 
-// NewAuthorizationInterceptorProviderWithToken returns a new provider function which, when invoked, returns an
-// interceptor which sets the provided auth token into the request header.  This is used for registry providers where
-// the token is known at provider creation (i.e. when logging in and explicitly pasting a token into stdin
-//
-// Note that the interceptor returned from this provider is always applied LAST in the series of interceptors added to
-// a client.
-func NewAuthorizationInterceptorProviderWithToken(token string) func(string) connect.UnaryInterceptorFunc {
-	return func(address string) connect.UnaryInterceptorFunc {
-		interceptor := func(next connect.UnaryFunc) connect.UnaryFunc {
-			return connect.UnaryFunc(func(
-				ctx context.Context,
-				req connect.AnyRequest,
-			) (connect.AnyResponse, error) {
-				if token != "" {
-					req.Header().Set(AuthenticationHeader, AuthenticationTokenPrefix+token)
-				}
-				return next(ctx, req)
-			})
+// TokenSet is used to provide authentication token in NewAuthorizationInterceptorProvider
+type TokenSet struct {
+	envKey          string
+	bufToken        string
+	remoteUsernames map[string]string
+	remoteTokens    map[string]string
```

Comment:
Instead of two maps, make it one map with a struct.

```go
type authkey struct {
  Username string
  Token string
}

type TokenSet struct {
  // ...
  tokens map[string]authkey
}
```

The neat part is you can break apart your parsing code since it's a dedicated type. If you were to follow the encoding package's form, it'd look like:

```go
func (ak *authkey) UnmarshalString(s string) error {
  ak.Username, ak.Token, found := strings.Cut(s, ":")
  if !found {
    return errors.New("invalid user:token format")
  }
  return nil
}
```